addons/hr_holidays/models/hr_leave_accrual_plan.py

```diff
@@ -9,16 +9,24 @@ class AccrualPlan(models.Model):
     _description = "Accrual Plan"
 
     name = fields.Char('Name', required=True)
-    time_off_type_id = fields.Many2one('hr.leave.type', string="Time Off Type")
+    time_off_type_id = fields.Many2one('hr.leave.type', string="Time Off Type",
+        help="""Specify if this accrual plan can only be used with this Time Off Type.
+                Leave empty if this accrual plan can be used with any Time Off Type.""")
     employees_count = fields.Integer("Employees", compute='_compute_employee_count')
-    level_ids = fields.One2many('hr.leave.accrual.level', 'accrual_plan_id')
+    level_ids = fields.One2many('hr.leave.accrual.level', 'accrual_plan_id', copy=True)
     allocation_ids = fields.One2many('hr.leave.allocation', 'accrual_plan_id')
     transition_mode = fields.Selection([
         ('immediately', 'Immediately'),
         ('end_of_accrual', "After this accrual's period")],
         string="Level Transition", default="immediately", required=True,
         help="""Immediately: When the date corresponds to the new level, your accrual is automatically computed, granted and you switch to new level
                 After this accrual's period: When the accrual is complete (a week, a month), and granted, you switch to next level if allocation date corresponds""")
+    level_count = fields.Integer('Levels', compute='_compute_level_count')
+
+    @api.depends('level_ids')
+    def _compute_level_count(self):
```

Comment:
Note that you could use a read_group, notably if level_count is used on a kanban view with lots of records for example to display this could be more efficient.

Reply:
Indeed